mrp_packaging_default/models/stock_move.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+
+from odoo import api, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    @api.model
+    def _packaging_vals_from_bom_line(self, vals):
+        """Fill vals with packaging info from BoM line."""
+        try:
+            bom_line = self.env["mrp.bom.line"].browse(vals["bom_line_id"])
+        except KeyError:
+            # No BoM line, nothing to do
+            return
+        vals.update(
+            {
+                "product_packaging_id": bom_line.product_packaging_id.id,
+                "product_packaging_qty": bom_line.product_packaging_qty,
+            }
+        )
```

Comment:
```suggestion
        """Fill vals with packaging info from BoM line."""
        bom_line_id = vals.get("bom_line_id")
        if bom_line_id:
            bom_line = self.env["mrp.bom.line"].browse(vals["bom_line_id"])
            vals.update(
                {
                    "product_packaging_id": bom_line.product_packaging_id.id,
                    "product_packaging_qty": bom_line.product_packaging_qty,
                }
            )
```
Using an if is usually nicer than a try except block :wink:

Reply:
This is the classical EAFP vs LBYL battle. People coming from langs such as C are more used to LBYL style. However, Python docs encourage EAFP. Thus, I'll leave as it was, because the end result is just the same but EAFP is generally considered more pythonic.

See https://stackoverflow.com/a/11360880/1468388 for a brief explanation.

Thanks for the review anyway! 🙂

Reply:
Thanks for the pointer, didn't know about this. Anyway I found [this article](https://realpython.com/python-lbyl-vs-eafp/) more telling than the stackoverflow thread, so linking it here FTR.